### PR TITLE
Add "REVERSE" / "REPEAT" / "RIGHT" / "LEFT" functions

### PR DESCRIPTION
**Pull request by asdf2014**
### Description

Add several string-related functions in this PR.

### Design

* Add left / right / repeat / reverse Druid expression functions
* Add LEFT / RIGHT / REPEAT / REVERSE Druid SQL functions

### CODE REVIEW

File: core/src/main/java/org/apache/druid/java/util/common/StringUtils.java [hunk 1]
```diff
@@ -360,4 +361,54 @@ public static String encodeBase64String(byte[] input)
...
+    if (len == 0 || count == 0) {
+      return "";
+    }
+    if (len == 1) {
+      final byte[] single = new byte[count];
+      Arrays.fill(single, value[0]);
+      return new String(single, StandardCharsets.UTF_8);
+    }
+    if (Integer.MAX_VALUE / count < len) {
+      throw new OutOfMemoryError();
```
Comment (egor-ryashin) [inline core/src/main/java/org/apache/druid/java/util/common/StringUtils.java]: I believe "there is a strong convention that errors are reserved for use by the JVM to indicate resource deficiencies, invariant failures, or other conditions that make it impossible to continue execution."

Comment (asdf2014) [reply core/src/main/java/org/apache/druid/java/util/common/StringUtils.java]: @egor-ryashin Thanks, fixed.

File: core/src/main/java/org/apache/druid/java/util/common/StringUtils.java [hunk 2]
```diff
@@ -399,7 +399,13 @@ public static String repeat(String s, int count)
       return new String(single, StandardCharsets.UTF_8);
     }
     if (Integer.MAX_VALUE / count < len) {
-      throw new OutOfMemoryError();
+      String errMsg = StringUtils.format(
```
Comment (egor-ryashin) [inline core/src/main/java/org/apache/druid/java/util/common/StringUtils.java]: I would go for a concise one `The produced string is too large.` I think a general Druid user is smart enough to understand it right away.

Comment (asdf2014) [reply core/src/main/java/org/apache/druid/java/util/common/StringUtils.java]: @egor-ryashin Done.

**Comment by asdf2014** [comment]
@egor-ryashin Do you have more comments?

File: core/src/main/java/org/apache/druid/java/util/common/StringUtils.java [hunk 3]
```diff
@@ -360,4 +361,54 @@ public static String encodeBase64String(byte[] input)
   {
     return BASE64_DECODER.decode(input);
   }
+
+  /**
+   * Returns a string whose value is the concatenation of the
+   * string {@code str} repeated {@code count} times.
+   * <p>
+   * If count or length is zero then the empty string is returned.
```
Comment (QiuMM) [inline core/src/main/java/org/apache/druid/java/util/common/StringUtils.java]: nit: If count or the length of the string...

File: core/src/main/java/org/apache/druid/math/expr/Function.java [hunk 4]
```diff
@@ -326,7 +343,6 @@ protected ExprEval eval(double param)
     }
   }
 
-
```
Comment (QiuMM) [inline core/src/main/java/org/apache/druid/math/expr/Function.java]: nit: remove this blank line.

File: core/src/test/java/org/apache/druid/java/util/common/StringUtilsTest.java [hunk 5]
```diff
@@ -160,4 +165,24 @@ public void testURLEncodeSpace()
...
   }
+
+  @Test
+  public void testRepeat()
+  {
+    {
+      Assert.assertEquals("", StringUtils.repeat("foo", 0));
+      Assert.assertEquals("foo", StringUtils.repeat("foo", 1));
+      Assert.assertEquals("foofoofoo", StringUtils.repeat("foo", 3));
+    }
```
Comment (QiuMM) [inline core/src/test/java/org/apache/druid/java/util/common/StringUtilsTest.java]: nit: maybe replace `}` with a blank line? There is no such code style in druid. The same for the `{` and `}` below.

File: core/src/main/java/org/apache/druid/java/util/common/StringUtils.java (hunk 3 above)
Comment (asdf2014) [reply core/src/main/java/org/apache/druid/java/util/common/StringUtils.java]: @QiuMM Thanks :+1:

Comment (asdf2014) [reply core/src/main/java/org/apache/druid/java/util/common/StringUtils.java]: Fixed.

File: core/src/main/java/org/apache/druid/math/expr/Function.java (hunk 4 above)
Comment (asdf2014) [reply core/src/main/java/org/apache/druid/math/expr/Function.java]: Removed.

File: core/src/test/java/org/apache/druid/java/util/common/StringUtilsTest.java (hunk 5 above)
Comment (asdf2014) [reply core/src/test/java/org/apache/druid/java/util/common/StringUtilsTest.java]: Fixed.

**Comment by asdf2014** [comment]
@QiuMM Thanks for your comments :+1: They have all been fixed.

**Pull request merged**